### ISSUE TITLE
[Phase 7] Square / LogSquare link end-to-end with sign convention

### ISSUE THREAD

**Issue by arob5** [arob5/sabi#69]
## Summary

Land square link end-to-end: numerically-stable `Square` and `LogSquare` `Map` implementations, sign-convention handling via `NonNegative` constraint, the non-central χ² closed-form pushforward through Gaussians, and a benchmark validating against the softplus / Identity baselines. This is the second of issue #55's alternative-link follow-throughs.

Design references:
- [`docs/link_functions.md` §3.2](../blob/main/docs/link_functions.md#32-concrete-maps) — the `Square` and `LogSquare` Map definitions.
- [`docs/link_functions.md` §4.2](../blob/main/docs/link_functions.md#42-what-the-emulator-emits-in-each-case) — under square link, the emulator emits `√lik` with a sign convention.
- [`docs/link_functions.md` §5](../blob/main/docs/link_functions.md#5-pushforward-dispatch) — `(Square, Normal)` is closed form (non-central χ²); `(Square, MVN)` is MC.
- [`docs/link_functions.md` §10 Q1](../blob/main/docs/link_functions.md#10-open-questions) — multimodal sign convention.

### Map subclass implementations

- `Square(z) = z²` — straightforward; closed form on Gaussians is non-central χ² with `df=1`, noncentrality parameter `λ = (μ/σ)²`.
- `LogSquare(z) = 2 log|z|` — naive `2 * jnp.log(jnp.abs(z))` has a singularity at `z = 0`. Use `2 * jnp.log(jnp.maximum(jnp.abs(z), eps))` or a piecewise stable form. Document the singularity behaviour in the Map's docstring and tie it to the `NonNegative` constraint that consumers should pin.

### Sign convention

`Square` density is invariant under `z ↦ -z`, so `y` is identifiable only up to a global sign per disconnected support component. The `NonNegative` constraint on `DensityDecomposition` is the interface for pinning the convention:

```python
DensityDecomposition(
    target_single=…,
    output_shape=(),
    link=LogSquare(),
    shift=LogProb(modeling_prior),
    constraint=NonNegative(),
)
```

Per `link_functions.md` §10 Q1: in the multimodal case, `NonNegative` everywhere effectively brings back the constraint we were trying to avoid. For the first cut, restrict square-link benchmarks to single-mode or simply-connected supports; document the multimodal failure mode in the tutorial notebook (Phase 8).

### Pushforward dispatch

Register:
- `(Square, Normal(loc=μ, scale=σ))` → scaled non-central χ² with `df=1`, `noncentrality=(μ/σ)²`. Closed form for moments (mean, variance) and density.
- `(Square, MultivariateNormal)` → MC. Element-wise square of correlated Gaussians is a correlated quadratic form; no clean joint closed form.
- `(LogSquare, Normal)` → MC.
- `(LogSquare, MultivariateNormal)` → MC.

The closed-form `(Square, Normal)` entry needs a sabi or ProbPipe `NonCentralChiSquared` distribution (or scaled equivalent). If neither exists yet, this PR adds a minimal `NonCentralChiSquared` distribution with `mean`, `variance`, `_log_prob`, and `_sample` per ProbPipe conventions.

### Benchmark / regression test

`tests/test_square_link.py`:
- Single-mode 2D synthetic posterior (e.g., banana, Gaussian) with closed-form analytical density.
- `DensityDecomposition(target_single=…, output_shape=(), link=LogSquare(), shift=LogProb(prior), constraint=NonNegative())`.
- Verify posterior recovery against analytical density (MMD or similar metric).
- Compare against `link=Identity()` and `link=LogSoftplus()` baselines on the same problem.

## Acceptance criteria

- [ ] `Square` and `LogSquare` Map subclasses pass numerical stability tests (no NaN/Inf at `z = 0` for the deliberately constrained-away path; agreement with reference at finite `z`).
- [ ] `LogSquare` docstring documents the singularity at `z = 0` and points to `NonNegative` constraint.
- [ ] `NonCentralChiSquared` distribution lands (in sabi if not already in ProbPipe) with `_log_prob`, `_sample`, `mean`, `variance`.
- [ ] Pushforward dispatch entries:
  - [ ] `(Square, Normal)` → `NonCentralChiSquared` (closed form).
  - [ ] `(Square, MultivariateNormal)` → MC fallback.
  - [ ] `(LogSquare, Normal)` and `(LogSquare, MultivariateNormal)` → MC fallback.
- [ ] `tests/test_square_link.py` end-to-end test:
  - [ ] Constructs square-link decomposition with `NonNegative` constraint.
  - [ ] Runs algorithm to convergence on a single-mode 2D problem.
  - [ ] Recovers posterior to within tolerance vs. analytical density.
  - [ ] Compares against `Identity` and `LogSoftplus` baselines.
- [ ] Full test suite passes.
- [ ] Sphinx build clean.
- [ ] Multimodal failure mode noted in tutorial-notebook queue for Phase 8.

## Out of scope

- Multimodal sign convention beyond `NonNegative`-everywhere — defer; no clean answer per `link_functions.md` §10 Q1.
- Gauss–Hermite or alternative quadrature for nonlinear-Map joint MVN pushforward — same MC fallback as softplus is fine.
- Acquisition variants exploiting square-link χ² structure — Phase 9.

## Dependencies

- #YY (Map + pushforward infra) — Maps and dispatch.
- #YY (DensityDecomposition split) — host class with `constraint` field.
- Phase 6 (softplus link) — same regression test pattern; can land in either order, but pattern reuse helps.

## Downstream

- Phase 8 (bridging tutorial) consumes this for the square-link tutorial section.
- Phase 9 (link-aware acquisition audit) uses square as one of the test cases.
